### PR TITLE
janet: new recipe

### PR DESCRIPTION
**Pull request by Crestwave**

### CODE REVIEW

File: dev-lang/janet/janet-0.5.0.recipe [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+libVersion="$portVersion"
+libVersionCompat="$libVersion compat >= ${portVersion%%.*}"
+
+PROVIDES="
+	janet$secondaryArchSuffix = $portVersion
+	cmd:janet
+	lib:libjanet = $libVersionCompat
```
Comment (Begasus) [inline dev-lang/janet/janet-0.5.0.recipe]: If you install in in $binDir and $libDir, shouldn't these contain $secondaryArchSuffix for janet_x86 on gcc2h?

File: dev-lang/janet/janet-0.5.0.recipe [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	devel:libjanet = $libVersionCompat
+"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	cmd:git
```
Comment (Begasus) [inline dev-lang/janet/janet-0.5.0.recipe]: Sort:
```diff
-	cmd:make
-	cmd:git
+	cmd:git
+	cmd:make
```

File: dev-lang/janet/janet-0.5.0.recipe (hunk 1 above)
Comment (Crestwave) [reply dev-lang/janet/janet-0.5.0.recipe]: Yes, I forgot to add the secondary architecture stuff, which is why I closed it; guess that I didn't do it in time.

Comment (Begasus) [reply dev-lang/janet/janet-0.5.0.recipe]: Why close it, you could push new changes here :)

**Comment by Begasus** [comment]
While you are fixing, could you create a patchset from the patch? :)

File: dev-lang/janet/janet-0.5.0.recipe (hunk 1 above)
Comment (Crestwave) [reply dev-lang/janet/janet-0.5.0.recipe]: I know; I just didn't want anyone reviewing the mistakes while I worked it out and tested it. It should be fine now, I think.

**Comment by Crestwave** [comment]
> While you are fixing, could you create a patchset from the patch? :)

It's already upstreamed, though.

**Comment by Crestwave** [comment]
Hmm... apparently I can't reopen my pull request. Can you do so or will I open another one?

**Comment by Crestwave** [comment]
#3864

**Comment by Begasus** [comment]
> > While you are fixing, could you create a patchset from the patch? :)
> 
> It's already upstreamed, though.

Ok, that could be fixed in the next release then (had a look but couldn't reopen myself

**Pull request closed without merging**